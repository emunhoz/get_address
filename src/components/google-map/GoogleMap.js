import React from 'react'
import PropTypes from 'prop-types'
import { Map, Marker, GoogleApiWrapper } from 'google-maps-react'

const GoogleMap = ({ google, lat = -23.5168338, lng = -46.6139824 }) => (
  <Map
    google={google}
    zoom={16}
    initialCenter={{ lat, lng }}
    center={{ lat, lng }}
  >
    <Marker title='Minha localização' name='location' position={{ lat, lng }} />
  </Map>
)

GoogleMap.propTypes = {
  google: PropTypes.object,
  lat: PropTypes.number,
  lng: PropTypes.number,
  geoCode: PropTypes.object
}

export default GoogleApiWrapper({
  apiKey: process.env.REACT_APP_GOOGLE_API_KEY
})(GoogleMap)
