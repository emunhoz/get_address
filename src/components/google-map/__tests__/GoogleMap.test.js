import React from 'react'
import { render } from '@testing-library/react'

import GoogleMap from '../GoogleMap'

describe('Component: GoogleMap', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(<GoogleMap />)
  })

  it('render the component', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
