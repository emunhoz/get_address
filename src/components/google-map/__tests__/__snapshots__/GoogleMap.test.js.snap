// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GoogleMap render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        Loading...
      </div>
    </div>
    <script
      src="https://maps.googleapis.com/maps/api/js?key=AIzaSyA6WLiraDtwQIUCYmhgnFjAam5dI-rPVMA&callback=loaderCB01580792112190&libraries=places&v=3&language=en"
      type="text/javascript"
    />
  </body>,
  "container": <div>
    <div>
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
